library/src/main/java/org/xmtp/android/library/messages/MessageV2.kt

```diff
@@ -18,7 +19,12 @@ typealias MessageV2 = org.xmtp.proto.message.contents.MessageOuterClass.MessageV
 
 class MessageV2Builder {
     companion object {
-        fun buildFromCipherText(headerBytes: ByteArray, ciphertext: CipherText?): MessageV2 {
+        fun buildFromCipherText(
+            headerBytes: ByteArray,
+            ciphertext: CipherText?,
+            senderHmac: ByteArray?,
+            shouldPush: Boolean,
```

Comment:
So unfortunately android doesn't make it as easy to extend MessageV2 with new fields like they can in iOS and Swift. 
I'm going to push up a commit with what I think my be necessary.